UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -340,11 +345,9 @@ public int GetStepCount()
         /// </summary>
         public void ResetReward()
         {
-            m_Reward = 0f;
-            if (m_Done)
-            {
-                m_CumulativeReward = 0f;
-            }
+            Debug.Assert(rewardProvider != null, "m_RewardProviderComponent is null and " +
+                "legacy method 'ResetReward' was called.");
```

Comment:
I wonder if `legacy` is the right word. As per our discussion, this might stay for a while.